### PR TITLE
Fix: broken links and typos

### PR DESCRIPTION
**Pull request by SachinSahu431**
### Description
Fix broken links and typographical errors in the existing readme files. 

For example [this section](https://github.com/opensearch-project/opensearch-build/blob/main/RELEASE_PROCESS_OPENSEARCH.md#sample-test-manifests) refers to [opensearch-2.8.0-test.yml](https://github.com/opensearch-project/opensearch-build/blob/main/manifests/2.8.0/opensearch-2.8.0-test.yml) file, which is no longer available on main branch, and has been moved to [opensearch-2.8.0 ](https://github.com/opensearch-project/opensearch-build/blob/opensearch-2.8.0/manifests/2.8.0/opensearch-2.8.0.yml) branch.

There are also some other changes such as removing clearly noticeable typos, unclosed brackets, adding missing link to Slack public release channel, etc.

### Issues Resolved
None

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: MAINTAINERS.md [hunk 1]
```diff
@@ -96,7 +96,7 @@ We currently use ZenHub only with our public and opensource repositories.
 ### Setting up ZenHub
 
 ZenHub can be easily added as an [extension]((https://www.zenhub.com/extension)) to chrome and firefox which can be downloaded for free from the ZenHub website. 
-Alternatively, we can use a the ZenHub [webapp link](https://app.zenhub.com/workspaces/engineering-effectiveness-614cf4272a385f0015d2b48f/board?repos=357723952,406037663) to view the board.
+Alternatively, we can use the ZenHub [webapp link](https://app.zenhub.com/workspaces/engineering-effectiveness-614cf4272a385f0015d2b48f/board?repos=357723952,406037663) to view the board.
```
Comment (gaiksaya) [inline MAINTAINERS.md:99]: Zenhub is not being used anymore by this project. Can you please remove line 87 to 99?
I believe we have an image in root of the repo related to zenhub that can be removed too.

File: MAINTAINERS.md [hunk 2]
```diff
@@ -117,16 +117,16 @@ If you are using the webapp - here is the [link](https://app.zenhub.com/workspac
...
-Description for each pipeline can also be found on the sprint board by clicking on the 3 dots next to the pipleine name.
+Description for each pipeline can also be found on the sprint board by clicking on the 3 dots next to the pipeline name.
 
 ### Creating Issues
 
 Follow the steps below to create issues on ZenHub workflow -
 
 1. Create the issue for the desired repository following the required guidelines for mandatory and optional fields on GitHub.
-2. Add an acceptance criteria for the issue
+2. Add acceptance criteria for the issue
```
Comment (gaiksaya) [inline MAINTAINERS.md:127]: `an` is right here I believe?

File: ONBOARDING.md [hunk 3]
```diff
@@ -75,7 +75,7 @@ See https://github.com/opensearch-project/opensearch-build/issues/1234 for detai
...
 1. **Adding tests:** Each library has a respective library tester associated with it that can be used to test you jenkins workflow. This tests can be used to verify that the workflow is making the calls. The build system used is gradle. 
-For example, this [PublishToNpm test](https://github.com/opensearch-project/opensearch-build-libraries/blob/main/tests/jenkins/TestPublishToNpm.groovy) uses [PublishToNpmLibTester](https://github.com/opensearch-project/opensearch-build-libraries/blob/main/tests/jenkins/lib-testers/PublishToNpmLibTester.groovy) with expected parameter that can be unique to your workflow. The assertions makes sure that calls to npm registry is made which is mandatory to release an artifact.
+For example, this [PublishToNpm test](https://github.com/opensearch-project/opensearch-build-libraries/blob/main/tests/jenkins/TestPublishToNpm.groovy) uses [PublishToNpmLibTester](https://github.com/opensearch-project/opensearch-build-libraries/blob/main/tests/jenkins/lib-testers/PublishToNpmLibTester.groovy) with expected parameter that can be unique to your workflow. The assertions make sure that calls to npm registry is made which is mandatory to release an artifact.
```
Comment (gaiksaya) [inline ONBOARDING.md:81]: I think `makes` was right here?

**Review by gaiksaya — CHANGES_REQUESTED** [review]
Thanks for fixing the documentation and spell checks. Some minor nits. Other wise looks good.

File: MAINTAINERS.md (hunk 1 above)
Comment (SachinSahu431) [reply MAINTAINERS.md]: I have removed all the irrelevant details related to Zenhub (lines 87 to 173).

**Review by gaiksaya — APPROVED** [review]
LGTM! Thanks!

**Comment by gaiksaya** [comment]
Hi @SachinSahu431 I am merging this for now. You can try [rebasing](https://git-scm.com/book/en/v2/Git-Branching-Rebasing) the git commits squash to 1. Makes it cleaner.
Also if you are interested in improving the documentation more, we have an open issue to collect all documentation in one place and update the Gh wiki for this repo. See https://github.com/opensearch-project/opensearch-build/issues/3936 for more details.
Thanks!

**Pull request merged**